src/Akka.Streams.Kafka.Tests/BugFix240SupervisionStrategy.cs

```diff
@@ -0,0 +1,434 @@
+                    if (counter % 7 == 0)
+                        throw new Exception("BOOM!");
+                    return t;
+                })
+                .SelectAsync(1, async elem =>
+                {
+                    await elem.CommitableOffset.Commit();
+                    return elem.Record.Value;
+                })
+                .ToMaterialized(this.SinkProbe<string>(), Keep.Right)
+                .Run(Materializer);
+
+            var messages = new List<string>();
+            probe.Request(11);
+            for (var i = 0; i < 6; i++)
+            {
+                messages.Add(probe.ExpectNext(TimeSpan.FromSeconds(5))); 
+            }
+
+            // stream fails at index 7
+            var err = probe.ExpectEvent();
+            err.Should().BeOfType<TestSubscriber.OnError>();
+            var exception = ((TestSubscriber.OnError)err).Cause;
+            exception.Message.Should().Be("BOOM!");
+
+            // stream should be dead here
+            probe.ExpectNoMsg(TimeSpan.FromSeconds(5));
+            probe.Cancel();
+            
+            // restart dead stream
+            probe = KafkaConsumer.CommittableSource(consumerSettings, Subscriptions.AssignmentWithOffset(new TopicPartitionOffset(topicPartition, Offset.Unset)))
+                .Select(t =>
+                {
+                    counter++;
+                    if (counter % 7 == 0)
+                        throw new Exception("BOOM!");
+                    return t;
+                })
+                .SelectAsync(1, async elem =>
+                {
+                    await elem.CommitableOffset.Commit();
+                    return elem.Record.Value;
+                })
+                .ToMaterialized(this.SinkProbe<string>(), Keep.Right)
+                .Run(Materializer);
+            
+            probe.Request(11);
+            for (var i = 0; i < 5; i++)
+            {
+                messages.Add(probe.ExpectNext(TimeSpan.FromSeconds(5))); 
+            }
+            probe.Cancel();
+
+            // end result should be gapless
+            messages.Select(s => int.Parse(s)).Should().BeEquivalentTo(Enumerable.Range(1, 11));
+        }        
+        
+        [Fact]
+        public async Task SupervisionStrategy_Decider_on_complex_stream_should_work()
+        {
+            var topic = CreateTopic(1);
+            var group = CreateGroup(1);
+            var topicPartition = new TopicPartition(topic, 0);
+            var callCount = 0;
+
+            // create a custom Decider with a "Restart" directive in the event of DivideByZeroException
+            Directive Decider(Exception cause)
+            {
+                callCount++;
+                return Directive.Resume;
+            }
+
+            var committerSettings = CommitterSettings.Create(Sys);
+            var consumerSettings = CreateConsumerSettings<string>(group);
+            var counter = 0;
+
+            await Source.From(Enumerable.Range(1, 11))
```

Comment:
Add a comment here explaining that this is part of the "arrange" stage of the test to help populate data we will read during our "act" stage

Reply:
Done